crawler/main.py

```diff
@@ -0,0 +1,68 @@
+import logging
+import logging.config
+from typing import Dict
+
+from crawler.config.logging import LOGGING_CONF  # type: ignore
+from crawler.config_helpers import get_centre_details, get_config
+from crawler.constants import COLLECTION_IMPORTS, COLLECTION_SAMPLES
+from crawler.db import (
+    copy_collection,
+    create_import_record,
+    create_mongo_client,
+    get_mongo_collection,
+    get_mongo_db,
+)
+from crawler.helpers import download_csv, parse_csv
+
+logging.config.dictConfig(LOGGING_CONF)
+logger = logging.getLogger(__name__)
+
+
+def run(sftp: bool, test_db_config: Dict = None) -> None:
+    if test_db_config is None:
+        test_db_config = {}
+
+    try:
+        config = get_config(test_db_config)
+        centres = get_centre_details(config)
+
+        with create_mongo_client(config) as client:
+            db = get_mongo_db(config, client)
+            imports_collection = get_mongo_collection(db, COLLECTION_IMPORTS)
+            samples_collection = get_mongo_collection(db, COLLECTION_SAMPLES)
+
+            if samples_collection.estimated_document_count() > 0:
+                logger.info(f"{COLLECTION_SAMPLES} collection is not empty so creating copy")
+                copy_collection(db, samples_collection)
+
+                logger.info(f"Removing all documents from {COLLECTION_SAMPLES}")
+                result = samples_collection.delete_many({})
+                logger.debug(f"{result.deleted_count} records deleted")
```

Comment:
Why do we archive the old samples in a different collection by timestamp? Is not something that could be equivalent to keeping them with their import date timestamp?

Reply:
Ok, when we know better what is going to be in the inputs we can decide

Reply:
Need to double check with Rich.